dnsimple/services.go

```diff
@@ -0,0 +1,93 @@
+package dnsimple
+
+import (
+	"fmt"
+)
+
+// ServicesService handles communication with the service related
+// methods of the DNSimple API.
+//
+// See https://developer.dnsimple.com/v2/services/
+type ServicesService struct {
+	client *Client
+}
+
+// ServiceSetting represents a single group of settings for a DNSimple Service.
+type ServiceSetting struct {
+	Name        string `json:"id,omitempty"`
+	Label       string `json:"label,omitempty"`
+	Append      string `json:"append,omitempty"`
+	Description string `json:"description,omitempty"`
+	Example     string `json:"example,omitempty"`
+	Password    bool   `json:"password,omitempty"`
+}
+
+// Service represents a Service in DNSimple.
+type Service struct {
+	ID               int              `json:"id,omitempty"`
+	Name             string           `json:"name,omitempty"`
+	ShortName        string           `json:"short_name,omitempty"`
+	Description      string           `json:"description,omitempty"`
+	SetupDescription string           `json:"setup_description,omitempty"`
+	RequiresSetup    bool             `json:"requires_setup,omitempty"`
+	DefaultSubdomain string           `json:"default_subdomain,omitempty"`
+	CreatedAt        string           `json:"created_at,omitempty"`
+	UpdatedAt        string           `json:"updated_at,omitempty"`
+	Settings         []ServiceSetting `json:"settings,omitempty"`
+}
+
+func servicePath(service interface{}) string {
```

Comment:
It's an `interface` because we want as output:

  - `"/services"` when `service` is `nil`
  - `"/services/1"` when `service` is `1` or `"1"`
  - `"/services/short_name"` when `service` is `"short_name"`